core/src/test/java/io/grpc/ServerProviderTest.java

```diff
@@ -49,6 +54,39 @@ public void unavailableProvider() {
     assertNull(ServerProvider.load(cl));
   }
 
+  @Test
+  public void contextClassLoaderProvider() {
+    ClassLoader ccl = Thread.currentThread().getContextClassLoader();
+    try {
+      // make a classloader that equals ServerProvider.class.getClassLoader()
+      // but that wont find any provider
+      ClassLoader cl = new ReplacingClassLoader(
+              getClass().getClassLoader(), serviceFile,
+              "io/grpc/ServerProviderTest-empty.txt") {
+        @Override
+        public boolean equals(Object obj) {
+          if (obj.equals(ServerProvider.class.getClassLoader())) {
+            return true;
+          }
+          return super.equals(obj);
+        }
+
+        @Override
+        public int hashCode() {
+          return super.hashCode();
+        }
+      };
+
+      // test that the context classloader is used as fallback
+      TrackResourceClassLoader tracker = new TrackResourceClassLoader(ccl);
```

Comment:
Instead of tracking which resources are loaded, how about using ReplacingClassLoader with ServerProviderTest-multipleProvider.txt (or similar) and asserting that the expected provider is returned?

Reply:
ok done :)